### ISSUE TITLE
Auto Deployment triggers multiple deploys

### ISSUE THREAD

**Issue by dblandin** [atmos/heaven#103]
We recently enabled Status webhook events to let Heaven handle auto-deployment of passing master builds but we've encountered some odd behavior and had to disable it pretty quickly. First, it seems like multiple deployments are triggered.

Here's a screenshot of my notification output after merging a PR containing just 1 commit:

![screen shot 2014-10-30 at 11 35 52 am](https://cloud.githubusercontent.com/assets/123595/4847430/d52868d8-6052-11e4-8038-e0dfe28858e6.png)

I'm not too familiar with the intent behind the `Deployment.latest_for_name_with_owner` class method but a deployment seems to trigger for each result returned.

We're deploying 3 apps trough heaven currently.

Here are some counts from `Deployment.latest_for_name_with_owner`:

```ruby
> Deployment.latest_for_name_with_owner('dscout/dscout').count
=> 4
> Deployment.latest_for_name_with_owner('dscout/dscout-web').count
=> 5
> Deployment.latest_for_name_with_owner('dscout/dscout-efflux').count
=> 6
```

Are multiple results expected?

**Comment by atmos** [comment]
Is this still a thing?

**Comment by dblandin** [comment]
Likely is. I haven't dared to enable this again without writing some specs.

**Comment by ajsharma** [comment]
I just turned the `Commit status updated from the API.` flag back on for our application's webhook and this happened to me too ( turned it back off right quick afterwards :tongue: )

**Comment by ajsharma** [comment]
Running the latest version of Heaven too

**Comment by filipegiusti** [comment]
On my case it's triggering two deployments to production because https://github.com/atmos/heaven/blob/master/app/models/deployment.rb#L7-L20 returns two Deployment, one for production and one for staging.

**Comment by filipegiusti** [comment]
What's the expected behavior for auto deploy with multiple environments? Deploy master to all of them or only to production?

**Comment by atmos** [comment]
At GitHub we had `auto-deploy enable hubot in <env>` and `auto-deploy disable hubot in <env>`. So you could configure different environments as they became available. There are slightly different rules for whether the user who last deployed pushed more commits and whether it was locked. I feel like maybe we need better docs of the behavior so we can clarify the intended behavior w/ the few variables.

**Comment by filipegiusti** [comment]
@atmos that behavior you described is not happening.

Have a look at https://github.com/atmos/heaven/blob/c94ab2f859218575b7e8992f5e38a33bfd3bb27c/app/models/commit_status.rb#L47-L50 then https://github.com/atmos/heaven/blob/master/app/models/deployment.rb#L7-L20 and finally https://github.com/atmos/heaven/blob/c94ab2f859218575b7e8992f5e38a33bfd3bb27c/app/models/auto_deployment.rb#L35-L38. Pay attention that on AutoDeployment it's not using the environment when creating a deployment, but CommitStatus triggers deployments for each environment.

**Comment by dblandin** [comment]
Looks like this was fixed with #158. Closing.
